[CORE-566] Add option to get workspace by Google Project Id to the support page

calypsomatic opened the pull request:
### Jira Ticket: https://broadworkbench.atlassian.net/browse/CORE-566

<!-- ### Dependencies --->
<!-- Include any dependent tickets and describe the relationship. Include any relevant Jira tickets. --->

## Summary of changes:
<!--Please give an abridged version of the ticket description here and/or fill out the following fields.-->

### What
- Uses the new rawls `getWorkspaceByGoogleProjectId` admin API to allow support to search for workspaces by GoogleProjectId.  Adds a third option on the workspace page.  Now support can search by workspace id, namespace/name OR google project id.

### Why
-

### Testing strategy
<!-- Note that changes impacting components in Storybook stories can be viewed at
https://www.chromatic.com/library?appId=65fc89c9335768720ff8605a&branch=<branch>
The branch corresponding to this PR is selected, and changes can be reviewed by commit. --->

- [ ] <!-- Test case 1 -->

<!-- ### Visual Aids -->
<!-- https://support.apple.com/guide/quicktime-player/record-your-screen-qtp97b08e666/mac -->

calypsomatic committed:
first pass

calypsomatic committed:
hopefully fix

calypsomatic commented on src/support/LookupSummaryAndPolicies.tsx:
```diff
@@ -34,6 +37,31 @@ export const LookupSummaryAndPolicies = (props: ResourceTypeSummaryProps) => {
     supportResources
   );
 
+  const supportSummaryProps: ResourceTypeSummaryProps = React.useMemo(() => {
```
This was tricky: the `loadSupportSummaryFn` is built into the `ResourceTypeSummaryProps`, and request to load the resource (in this case workspace) are handled there, so I couldn't just define a single `loadSupportSummaryFn` in the `SupportResourceType`.  I'm not super thrilled about this solution, but it seems to work.  Am open to better ideas.

davidangb reviewed (approved):
:+1:, I am fine with this as-is.

Brainstorming on your comment about not being thrilled at the solution:

* See #5337, where we added the lookup-by-name. This is where we added a `lookupResourceIdFn`. You could take the same route and add a new `lookupResourceIdByGoogleProjectOrWhatever` function.
* In hindsight, an architecturally clean solution would be to modify the existing `adminGetWorkspaceId` API to be a generic "find a workspace by name or project or some other field in the future` and then pass different query params to that API for lookup by project or name.

Again, I think this is ok as it is.

calypsomatic commented:
> * See #5337, where we added the lookup-by-name. This is where we added a `lookupResourceIdFn`. You could take the same route and add a new `lookupResourceIdByGoogleProjectOrWhatever` function.

The annoying thing here is that the new rawls API returns the workspace directly (since there already existed an underlying method to get the workspace by google project id), so I was able to skip the roundabout way of getting the workspace id and THEN querying for the workspace, but it made it not quite work to do that same thing as `lookupResourceIdFn`.
